kong/router/fields.lua

```diff
@@ -213,6 +214,24 @@ if is_http then
   local HTTP_SEGMENTS_OFFSET = 1
 
 
+  local function get_http_segments(params)
+    if not params.segments then
+      HTTP_SEGMENTS_REG_CTX.pos = 2 -- reset ctx, skip first '/'
+      params.segments = re_split(params.uri, "/", "jo", HTTP_SEGMENTS_REG_CTX)
+    end
+
+    return params.segments
+  end
+
+
+  FIELDS_FUNCS["http.path.segments.len"] =
+  function(params)
```

Comment:
seems that we always use the following style in everywhere of kong

```
<function variable / ...> = function           -- function key word is in the assignment line
...
end
```

Reply:
Here we add a function to the hash table, it is not a normally function definition.

And sometimes `local f = function()... end` may show the intention that the `f` is overwrited.